lib/rack/mock.rb

```diff
@@ -210,7 +214,7 @@ def body
       #     ...
       #     res.body.should == "foo!"
       #   end
-      buffer = String.new
+      buffer = String.new.force_encoding(encoding)
```

Comment:
It's better to do `String.new(encoding: ...)`. But not sure about backwards compatibility.